File: Deepskilling/Module1/Design Patterns and Principles/Exercise 3/Single Responsibility Principle.cs
using System;

class Report
{
    public string Content;

    public Report(string content)
    {
        Content = content;
    }
}

class ReportPrinter
{
    public void PrintReport(Report report)
    {
        Console.WriteLine(report.Content);
    }
}

class Program
{
    static void Main()
    {
        Report report = new Report("Monthly Report");

        ReportPrinter printer = new ReportPrinter();

        printer.PrintReport(report);
    }
}

File: Deepskilling/Module1/Design Patterns and Principles/Exercise 3/Single Responsibility Principle.js
class Report{

    constructor(content){
        this.content = content;
    }

}

class ReportPrinter{

    print(report){
        console.log(report.content);
    }

}

let report = new Report("Monthly Report");

let printer = new ReportPrinter();

printer.print(report);
